activesupport/lib/active_support/message_pack.rb

```diff
@@ -0,0 +1,26 @@
+# frozen_string_literal: true
+
+begin
+  require "msgpack"
+rescue LoadError => error
+  $stderr.puts "You don't have msgpack installed in your application. " \
+    "Please add it to your Gemfile and run bundle install."
+  raise error
+end
+
+require_relative "message_pack/extensions"
+require_relative "message_pack/serializer"
+
+module ActiveSupport
+  module MessagePack
+    extend Serializer
+    self.message_pack_factory = Extensions.configure_factory(::MessagePack::Factory.new)
+
+    module CacheSerializer
+      extend Serializer
+      self.message_pack_factory = ActiveSupport::MessagePack.message_pack_factory.dup
```

Comment:
Why do you need to dup it?

Reply:
It is part of my plan to have `AS::MessagePack::CacheSerializer` share behavior with `AS::MessagePack`.  Later, I plan to override the handler for `Object` and `ActiveRecord::Base`.